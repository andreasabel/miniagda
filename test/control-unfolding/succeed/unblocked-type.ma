-- Level-polymorphic identity

fun TYPE : (i : Size) -> Set $i
{ TYPE i = Set i }

fun TPolyId : (i : Size) -> Set $i
unfold TYPE
{ TPolyId i = (A : TYPE i) -> A -> A }

fun polyId : (i : Size) -> TPolyId i
unfold TPolyId
{ polyId i A x = x }

-- Polymorphic identity

fun SET : Set 1
{ SET = Set }

fun TId0 : Set 1
unfold SET
{ TId0 = (A : SET) -> A -> A }

let SET0 : Set 1 = Set
let TId : Set 1 unfold SET0 = (A : SET0) -> A -> A

-- TODO
-- data Empty : unfold SET in SET
-- {}
