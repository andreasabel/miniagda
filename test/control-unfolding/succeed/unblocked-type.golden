--- opening "unblocked-type.ma" ---
--- scope checking ---
--- type checking ---
ty-u  TYPE : (i : Size) -> Set $i
{ TYPE i = Set i
}
ty-u  TPolyId : (i : Size) -> Set $i
{ TPolyId i = (A : TYPE i) -> A -> A
}
mixk  polyId : (i : Size) -> TPolyId i
{ polyId i A x = x
}
ty-u  SET : Set 1
{ SET = Set
}
ty-u  TId0 : Set 1
{ TId0 = .[A : SET] -> A -> A
}
univ  SET0 : Set 1
univ  SET0 = Set
type  TId : Set 1
type  TId = .[A : SET0] -> A -> A
--- evaluating ---
--- closing "unblocked-type.ma" ---
